bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/java/org/eclipse/smarthome/config/discovery/internal/AutomaticInboxProcessorTest.java

```diff
@@ -374,10 +380,85 @@ public void testThingWithConfigIsBeingRemoved() {
         assertThat(results.size(), is(1));
         assertThat(results.get(0).getThingUID(), is(equalTo(THING_UID2)));
 
-        inboxAutoIgnore.removed(thing2);
+        automaticInboxProcessor.removed(thing2);
 
         results = inbox.getAll();
         assertThat(results.size(), is(0));
     }
 
+    @Test
+    public void testAutomaticDiscoveryResultApprovalIfInboxEntriesAddedAfterApprovalPredicatesAreAdded() {
+        automaticInboxProcessor.addInboxAutoApprovePredicate(
+                discoveryResult -> discoveryResult.getThingTypeUID().equals(THING_TYPE_UID));
+
+        // The following discovery result is automatically approved, as it has matching thing type UID
+        inbox.add(DiscoveryResultBuilder.create(THING_UID).build());
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID)));
+
+        // The following discovery result is not automatically approved, as it does not have matching thing type UID
+        inbox.add(DiscoveryResultBuilder.create(THING_UID3).build());
+        verify(thingRegistry, never()).add(argThat(thing -> thing.getUID().equals(THING_UID3)));
+    }
+
+    @Test
+    public void testAutomaticDiscoveryResultApprovalIfInboxEntriesExistBeforeApprovalPredicatesAreAdded() {
+        inbox.add(DiscoveryResultBuilder.create(THING_UID).build());
+        inbox.add(DiscoveryResultBuilder.create(THING_UID2).build());
+        inbox.add(DiscoveryResultBuilder.create(THING_UID3).build());
+
+        // Adding this inboxAutoApprovePredicate will auto-approve the first two discovery results as they have matching
+        // thing type UID.
+        automaticInboxProcessor.addInboxAutoApprovePredicate(
+                discoveryResult -> discoveryResult.getThingTypeUID().equals(THING_TYPE_UID));
```

Comment:
Why does a result with ID `THING_UID2` match `THING_UID`?

Reply:
The predicate matches thing types UIDs, not thing UIDs; both `THING_UID` and `THING_UID2` have the same thing type UID.